### PR TITLE
Create outlined button component

### PR DESCRIPTION
**Pull request by piitaya**
## Proposed change

Create dedicated button component for new material outlined button.
It also fixes a bug with button padding in iOS.

## Type of change

- [ ] Dependency upgrade
- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [ ] The code change is tested and works locally.
- [ ] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/components/ha-outlined-button.ts [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import { styles } from "@material/web/button/lib/outlined-styles.css";
+import { styles as sharedStyles } from "@material/web/button/lib/shared-styles.css";
+import { MdOutlinedButton } from "@material/web/button/outlined-button";
+
+@customElement("ha-outlined-button")
+export class HaOutlinedButton extends MdOutlinedButton {
+  static override styles = [
+    sharedStyles,
+    styles,
```
Comment (bramkragten) [inline src/components/ha-outlined-button.ts:12]:
```suggestion
    ...MdOutlinedButton.styles
```

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/components/ha-outlined-button.ts:12]: We should do that anyway probably

[the remark this answers is not in the document]
Comment (piitaya) [reply src/components/ha-outlined-button.ts:12]: My first approach was to extends `Button` and `IconButton`. There is no OutlinedButton. `MdOutlinedButton` extends `Button`.

[the remark this answers is not in the document]
Comment (bramkragten) [reply src/components/ha-outlined-button.ts:12]: https://github.com/material-components/material-web/blob/main/button/outlined-button.ts#LL42C55-L42C55

[the remark this answers is not in the document]
Comment (piitaya) [reply src/components/ha-outlined-button.ts:12]: Oh there is `OutlinedButton` but no `OutlinedIconButton`🤦‍♂️

**Review by bramkragten — APPROVED** [review]

**Pull request merged**